### ISSUE TITLE
[Feature Request] Support for packages with multiple licenses

### ISSUE THREAD

**Issue by jcasner** [actions/dependency-review-action#263]
One of the packages we're importing has multiple licenses based on dependent projects. The Github dependency-graph API returns all 3 licenses. We have each of the 3 licenses in our `allow-list`, but because this is a single string response, it's failing the dependency review. 

Response from the depency-graph API:
```json
  {
    "change_type": "added",
    "manifest": "package-lock.json",
    "ecosystem": "npm",
    "name": "@fortawesome/fontawesome-free",
    "version": "5.15.4",
    "package_url": "pkg:npm/%40fortawesome/fontawesome-free@5.15.4",
    "license": "CC-BY-4.0 AND MIT AND OFL-1.1",
    "source_repository_url": "https://github.com/FortAwesome/Font-Awesome",
    "scope": "runtime",
    "vulnerabilities": []
  },
```

**Proposal**
For a situation like this, the action would parse the license field and use the operator (AND) to check that all 3 licenses are in our allow-list. I can see the possibility of a dual-licensed package including an `OR` (e.g.: `CC-BY-4.0 OR MIT`) and so we'd want to use the operator (AND or OR) to validate against the `allow-list` or `deny-list` appropriately. 

Thanks for your consideration!

**Comment by febuiles** [comment]
@jcasner Thanks for the report. This is something originally reported in https://github.com/actions/dependency-review-action/pull/131 that's actively being worked on using `spdx-satisfies` to evaluate these expressions. I'll update this issue once there are news to share!

**Comment by febuiles** [comment]
The latest release (`v3`) is fully SPDX-compliant and now has support for `AND`/`OR` expressions. Sadly, your specific example is a [known bug in an upstream library](https://github.com/jslicense/spdx-satisfies.js/issues/14), so I'm not confident the newest release will allow you to specific all three licenses as listed above. The `OR` expressions should work fine.

I'm leaving the bug report open until `spdx-satisfies` has been updated.

**Comment by JPLachance** [comment]
Hello!

I do have a very similar issue. We have a long `allow-list` and we are constantly facing issues like the one described here.

Do we have an ETA?

**Comment by febuiles** [comment]
@JPLachance we can't proceed until https://github.com/jslicense/spdx-satisfies.js/issues/14 is fixed upstream. All ears if you have suggestions on how to improve the parsing of SPDX expressions!

**Comment by prakyathr** [comment]
Any updates on this issue? We are using many packages which uses multiple packages. Since this is a paid service, can we expect any timeline to fix it?

**Comment by jonjanego** [comment]
> We are using many packages which uses multiple packages

HI @prakyathr thank you for following up on this. Could you clarify, do you mean you're using packages that use multiple licenses?

**Comment by prakyathr** [comment]
Hi @jonjanego , you are right. I meant packages that uses multiple licenses. I fixed the typo in the comment above as well. Anyway, will it be fixed soon?

**Comment by juxtin** [comment]
I just wanted to add a link to @elireisman comment here so it doesn't get lost: https://github.com/actions/dependency-review-action/pull/719/files#r1636935433

**Comment by jtomkiew-mng** [comment]
Just putting a note here that since version 4.3.4 there is no workaround for packages with licence expressions (#792) and implementing this would be a welcome addition.

**Comment by wmmc88** [comment]
@juxtin So since version 4.3.4, is the only way to pass the pipeline when some dependencies use `AND` licenses is to add the package to `allow-dependencies-licenses`? Since expressions in `allow-licenses` are no longer allowed

**Comment by jaymevillafranca** [comment]
This is still an issue as of v4.3.4. I opened a support ticket recently and they pointed me to this issue. Why hasn't there been a fix yet after 2 years?
![dr-multiplelicense-issue](https://github.com/user-attachments/assets/5b0be05a-4cd3-45e3-8026-d112662e90a4)

**Comment by jonjanego** [comment]
Thanks for the feedback. We don't have this in our immediate plans but it's on our radar.

**Comment by IronMitch** [comment]
Hi, has this request been reviewed since July?  I had a Microsoft developer reach out asking when it may be remediated

**Comment by jaymevillafranca** [comment]
When can we expect a fix? Dependency Review **still** isn’t accurately processing the allowlist for packages that use multiple licenses, which remains a significant issue.

**Comment by jeffpaul** [comment]
FWIW a workaround exists to include in your allowed licenses a concatenated list using AND in these cases.  Suboptimal for sure, but it works.

**Comment by Loki-Afro** [comment]
@jeffpaul thats not only suboptimal but in some cases not even viable
comming from an AGPL project where you can not put in any proprietary license

this one https://github.com/APIDevTools/json-schema-ref-parser has the following license
`JSON AND LicenseRef-scancode-proprietary-license AND MIT`

now i shall include `JSON, `LicenseRef-scancode-proprietary-license` and `MIT`, meaning that each one of them is viable?  simply not possible for an AGPL project

only way i see is using `allow-dependencies-licenses` for an exception for that package ...

**Comment by Loki-Afro** [comment]
@jeffpaul i dig deeper into the issue, can you confirm that your workaround still works in your usecase?

![Image](https://github.com/user-attachments/assets/c124bc9b-d7b4-40d7-85c2-15805bd1702c)

I can see some error being thrown that those expressions are not supported by the underlying spdx library and not treated as a plain string like your workaround suggested ...

**Comment by ahpook** [comment]
Note to those watching this issue - there is a change merged into `main` via #916 which should address the problem. If it's feasible in your environment, please run a test against it and see if your issues with multiple licenses are addressed.

**Comment by jtomkiew-mng** [comment]
> Note to those watching this issue - there is a change merged into `main` via [#916](https://github.com/actions/dependency-review-action/pull/916) which should address the problem. If it's feasible in your environment, please run a test against it and see if your issues with multiple licenses are addressed.

This basically fixes [the breaking change introduced in v4.3.4](https://github.com/actions/dependency-review-action/issues/792) - now you can again have SPDX expressions in allow/deny lists (yay!).

This does not however handle SPDX expressions in packages, based on allow/deny lists (i.e., a package with `Apache-2.0 AND MIT` licence expression will not be allowed if your allow-list is: `Apache-2.0, MIT`), which I believe this issue we are commenting on is about (sad panda).

Anyway, kudos to everyone involved in #916, we can finally use new action version!

**Comment by jebeaudet** [comment]
> This does not however handle SPDX expressions in packages, based on allow/deny lists (i.e., a package with `Apache-2.0 AND MIT` licence expression will not be allowed if your allow-list is: `Apache-2.0, MIT`), which I believe this issue we are commenting on is about (sad panda).

@jtomkiew-mng IANAL, but I answered on that subject here https://github.com/actions/dependency-review-action/pull/916#issuecomment-2816889099.
